src/routes/(auth)/forgot-password/+page.server.ts

```diff
@@ -0,0 +1,40 @@
+import type { Actions, PageServerLoad } from './$types';
+import { redirect, fail } from '@sveltejs/kit';
+import prisma from '@db';
+import { env } from '$env/dynamic/private';
+import { sendResetEmail } from '$lib/server/email/reset-password';
+
+export const load: PageServerLoad = async () => {
+	if (env.MAIL_ENABLED === 'false') {
+		throw redirect(303, '/');
+	}
+};
+
+export const actions: Actions = {
+	default: async ({ request }) => {
+		const data = await request.formData();
+
+		const usernameOrEmail = data.get('username-email');
+
+		if (!usernameOrEmail) {
+			return fail(400, { success: false, errors: ['All fields are required'] });
+		}
+
+		const user = await prisma.user.findFirst({
+			where: {
+				OR: [{ username: usernameOrEmail.toString() }, { email: usernameOrEmail.toString() }]
+			}
+		});
+
+		if (!user) {
+			return fail(400, { success: false, errors: ['Username or e-mail not found'] });
```

Comment:
I think this can be a slight security issue, since it will let people "find" which usernames or emails are in the server by brute forcing? Not a big issue, but most sites just have a vague "If you have an account, we will send you a verification link" type message. I think it would be better to do that.

Reply:
Good point, will remove any ways for users to "find" usernames/emails.

Reply:
Addressed in latest commit, how does it look?

Reply:
Looks good!